src/Components/Web/src/Virtualization/InfiniteScroll.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information. 
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Virtualization
+{
+    /// <summary>
+    /// Provides functionality for rendering a virtualized list from a source of unknown size.
+    /// </summary>
+    /// <typeparam name="TItem"></typeparam>
+    public sealed class InfiniteScroll<TItem> : ComponentBase
+    {
+        /// <summary>
+        /// Gets or sets the item template for the list.
+        /// </summary>
+        [Parameter]
+        public RenderFragment<TItem> Item { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the template for the footer of the list.
+        /// </summary>
+        [Parameter]
+        public RenderFragment<int> Footer { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the function retrieving items given a start index.
+        /// </summary>
+        [Parameter]
+        public ItemsProviderDelegate<TItem> ItemsProvider { get; set; } = default!;
+
+        /// <summary>
+        /// Gets or sets the item count requested to the <see cref="ItemsProvider"/>.
+        /// </summary>
+        [Parameter]
+        public int RequestedItemCount { get; set; } = 50;
+
+        /// <summary>
+        /// Gets the size of each item in pixels.
+        /// </summary>
+        [Parameter]
+        public float ItemSize { get; set; }
+
+        private int _itemCount;
+
+        /// <inheritdoc />
+        protected override void OnParametersSet()
+        {
+            if (Item == null)
+            {
+                throw new InvalidOperationException(
+                    $"{GetType()} requires template parameter '{nameof(Item)}' to be specified and non-null.");
+            }
+
+            if (Footer == null)
+            {
+                throw new InvalidOperationException(
+                    $"{GetType()} requires template parameter '{nameof(Footer)}' to be specified and non-null.");
```

Comment:
Hypothetically, if we wanted to make `Footer` optional, would that make sense at all? Would we just render no UI at the end, or would we need some kind of framework-supplied default?

Reply:
I've added a default which is just an empty div whose height matches `ItemSize`. I think this works fine for `Virtualize`, but it looks a little strange for `InfiniteScroll`. Maybe rendering no UI at the end would be nicer.

Reply:
That is, it might be nicer specifically for `InfiniteScroll` when no footer is provided.